Civi/Api4/Generic/Traits/ArrayQueryActionTrait.php

```diff
@@ -174,14 +174,15 @@ public static function filterCompare($row, $condition) {
         return !in_array($value, $expected);
 
       case 'CONTAINS':
+      case 'NOT CONTAINS':
         if (is_array($value)) {
-          return in_array($expected, $value);
+          return in_array($expected, $value) == ($operator == 'CONTAINS');
         }
         elseif (is_string($value) || is_numeric($value)) {
           // Lowercase check if string contains string
-          return strpos(strtolower((string) $value), strtolower((string) $expected)) !== FALSE;
+          return (strpos(strtolower((string) $value), strtolower((string) $expected)) !== FALSE) == ($operator == 'CONTAINS');
         }
-        return $value == $expected;
+        return ($value == $expected) == ($operator == 'CONTAINS');
```

Comment:
I'm also not sure how to get here. If it isn't an array and it isn't a string or numeric (which would include bools), then what is it?

Reply:
APIv4 supports pure booleans, which are not numeric. A value of `TRUE` or `FALSE` would throw an error if you tried to run them through `strtolower()` so this fallback catches those.

Reply:
Makes sense. I did see tests hitting that code, so I think it's good.